beetsplug/bpd/__init__.py

```diff
@@ -686,6 +758,13 @@ def run(self, conn):
                 for data in results:
                     yield conn.send(data)
 
+        except TypeError:
+            # The client provided too many arguments.
+            raise BPDError(ERROR_ARG,
+                           u'wrong number of arguments for "{}"'
+                           .format(self.name),
+                           self.name)
```

Comment:
I know this is a little tricky, but this `TypeError` handler may be a little too broad. It could catch simple programming errors inside of command-handler functions, like `'a' + 2`, for example. It might be best to use our `inspect` helper to explicitly check the argument count of `func`.

Reply:
Thanks for insisting on this point- it lead me to fixing the deeper issue of why bpd was crashing at all when there was a catch-all `except` clause. The answer was a bug in the line that logged the backtrace in that clause. Now we get `bpd.ERROR_SYSTEM` on unexpected exceptions with the proper log message instead of the server crashing!

Reply:
I think this bug has been there for a decade! https://github.com/beetbox/beets/commit/5ee460bcd5c05f6af0651dacbe47d67142018fa7#diff-88d200ed1c440335655e7351f6320344R494

Reply:
Oh wow! Nice catch. That is extremely depressing—I suddenly feel very old. 👴